control/controlhttp/client.go

```diff
@@ -28,18 +28,24 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"math"
 	"net"
 	"net/http"
 	"net/http/httptrace"
+	"net/netip"
 	"net/url"
+	"sort"
+	"sync/atomic"
 	"time"
 
 	"tailscale.com/control/controlbase"
+	"tailscale.com/envknob"
 	"tailscale.com/net/dnscache"
 	"tailscale.com/net/dnsfallback"
 	"tailscale.com/net/netutil"
 	"tailscale.com/net/tlsdial"
 	"tailscale.com/net/tshttpproxy"
+	"tailscale.com/tailcfg"
```

Comment:
I was going to say I thought Dave wanted this package to be kinda Tailscale-free and just a Noise thing, but I guess this was already a dep and we lost that cleanliness property some time ago:

```
bradfitz@book1pro controlhttp % ~/go/bin/deppath tailscale.com/control/controlhttp /tailcfg
["tailscale.com/control/controlhttp" "tailscale.com/net/dnsfallback" "tailscale.com/net/netns" "tailscale.com/net/interfaces" "tailscale.com/hostinfo" "tailscale.com/tailcfg"]
["tailscale.com/control/controlhttp" "tailscale.com/net/dnsfallback" "tailscale.com/tailcfg"]
["tailscale.com/control/controlhttp" "tailscale.com/net/netutil" "tailscale.com/net/interfaces" "tailscale.com/hostinfo" "tailscale.com/tailcfg"]
```